src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = {};
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        stateHistory.push({});
+        break;
+
+      case 'addProperties':
+        currentState = { ...(stateHistory.at(-1) || { ...state }) };
+
+        stateHistory.push(Object.assign(currentState, action.extraData));
```

Comment:
The `Object.assign` operation modifies `currentState` and then pushes it to `stateHistory`. Instead, modify `currentState` and push it outside of the `switch` block to adhere to the rule of pushing a copy at the end of each loop cycle.